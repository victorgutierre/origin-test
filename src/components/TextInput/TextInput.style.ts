import styled from 'styled-components';

export const TextInputWrapper = styled.div`
  width: 100%;
  margin-bottom: 2rem;

  @media screen and (min-width: 768px) {
    width: 60%;
    margin-right: 2rem;
  }
`;

export const Label = styled.p`
  display: block;
  margin-bottom: 0.5rem;
  color: #202a31;
  font-size: 1rem;
  text-align: left;
`;

export const TextInputBox = styled.div`
  position: relative;

  span {
    position: absolute;
    color: #cdd5db;
    font-size: 1.5rem;
    margin-left: 1rem;
    top: 50%;
    transform: translateY(-50%);
    z-index: 1;
  }

  input {
    width: 100%;
    height: 4rem;
    padding: 1rem 1rem 1rem 2.5rem;
    font-size: 1.5rem;
    font-weight: bold;
    border-radius: 8px;
    border: 1px solid #eaeef2;
    position: relative;
    outline: 0;
    color: #516473;
  }
`;
