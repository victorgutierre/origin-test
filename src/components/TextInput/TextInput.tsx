import React from 'react';
import { TextInputWrapper, Label, TextInputBox, Input } from './TextInput.style';

export const TextInput = () => {
  return (
    <TextInputWrapper>
      <Label>Total Amount</Label>

      <TextInputBox>
        <span>$</span>
        <Input type="text" />
      </TextInputBox>
    </TextInputWrapper>
  );
};
