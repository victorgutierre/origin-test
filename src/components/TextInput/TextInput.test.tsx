import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { render, fireEvent } from '@testing-library/react';
import { TextInput } from './TextInput';

describe('<TextInput />', () => {
  const amount = '25000';
  const setAmount = jest.fn();

  const component = <TextInput amount={amount} setAmount={setAmount} />;

  it('Should render the TextInput component', () => {
    const wrapper = shallow(component);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Text input currency must render', () => {
    const { queryByTestId } = render(component);
    const date = queryByTestId('text-input-currency');
    expect(date).toBeTruthy();
  });

  it('Change input value setAmount should be called', () => {
    const { queryByTestId } = render(component);
    const element = queryByTestId('text-input-currency');
    fireEvent.change(element, { target: { value: '3000000' } });
    expect(setAmount).toHaveBeenCalled();
  });

  it('When Change input value should be called', () => {
    const { queryByTestId } = render(component);
    const element = queryByTestId('text-input-currency');
    fireEvent.change(element, { target: { value: '3000000' } });

    expect(element.getAttribute('value')).toBe('30,000.00');
  });
});
