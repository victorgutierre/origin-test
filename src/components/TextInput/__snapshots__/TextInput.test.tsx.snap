// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> Should render the TextInput component 1`] = `
<styled.div>
  <styled.p>
    Total Amount
  </styled.p>
  <styled.div>
    <span>
      $
    </span>
    <CurrencyInput
      allowNegative={false}
      autoFocus={false}
      data-testid="text-input-currency"
      decimalSeparator="."
      inputType="tel"
      onChange={[Function]}
      onChangeEvent={[Function]}
      precision="2"
      prefix=""
      selectAllOnFocus={false}
      suffix=""
      thousandSeparator=","
      value={25000}
    />
  </styled.div>
</styled.div>
`;
